libexec/rtld-elf/aarch64/rtld_c18n_machdep.h

```diff
@@ -28,20 +28,50 @@
 #ifndef RTLD_C18N_MACHDEP_H
 #define RTLD_C18N_MACHDEP_H
 
+#define	C18N_TRUSTED_FRAME_SIZE		15
+
+#ifndef IN_ASM
 /*
  * Stack unwinding
  */
 struct trusted_frame {
 	void *fp;
 	void *pc;
+	/*
+	 * Address of the next trusted frame
+	 */
 	ptraddr_t next;
+	/*
+	 * Number of return value registers, encoded in enum tramp_ret_args
+	 */
 	uint8_t ret_args : 2;
+	/*
+	 * This field contains the code address in the trampoline that the
+	 * callee should return to. This is only used by unwinders to detect
+	 * compartment boundaries.
+	 */
 	ptraddr_t cookie : 62;
+	/*
+	 * INVARIANT: This field contains the top of the caller's stack when the
+	 * caller made the call.
+	 */
+	void *n_sp;
```

Comment:
Is the `n_sp` for `next_sp`? Given the surrounding code, it leads me to believe that this is `new_sp` as `o_sp` is right under it.

Reply:
It is indeed new sp